entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEvaluator.java

```diff
@@ -14,9 +14,11 @@ public class EntityChangeEvaluator {
   public static ChangeResult evaluateChange(
       Collection<Entity> existingEntities, Collection<Entity> updatedEntities) {
     Map<String, Entity> existingEntityMap =
-        existingEntities.stream().collect(Collectors.toMap(Entity::getEntityId, identity()));
+        existingEntities.stream()
+            .collect(Collectors.toMap(Entity::getEntityId, identity(), (v1, v2) -> v2));
```

Comment:
When would we have multiple entities with the same ID? isn't that a bug?

Reply:
not sure we are getting that from the clients. I have seen exception in upsert entities map which is defined below this. I have just added this to make it fail safe

Reply:
My concern is that making this lenient can hide a code bug. For example, if we're updating the same entity twice then that's likely a caller bug and may indicate a problem. Rather than making this fail safe, can we investigate where we've seen that and how we're getting there?

We can of course relax this if we determine it sometimes makes sense to see the same value twice, but we should root cause if it's an upstream issue first.

Reply:
i have seen it with agent module entity. Let me chase that code. bit unfamiliar with that part